store/src/main/java/com/nytimes/android/external/store/base/impl/BarcodeStoreBuilder.java

```diff
@@ -0,0 +1,110 @@
+package com.nytimes.android.external.store.base.impl;
+
+
+import com.nytimes.android.external.cache.Cache;
+import com.nytimes.android.external.store.base.DiskRead;
+import com.nytimes.android.external.store.base.DiskWrite;
+import com.nytimes.android.external.store.base.Fetcher;
+import com.nytimes.android.external.store.base.Parser;
+import com.nytimes.android.external.store.base.Persister;
+import com.nytimes.android.external.store.base.experimental.Store;
+import com.nytimes.android.external.store.util.NoopPersister;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import rx.Observable;
+import rx.functions.Func1;
+
+/**
+ * Builder where there parser is used.
+ */
+public class BarcodeStoreBuilder<Raw, Parsed, Key> {
+    private final List<Parser> parsers = new ArrayList<>();
```

Comment:
actually parser needs to be this way because you may be passing in 2 parsers, one of which is <String,Integer> and the other that is <Integer,Long> so the Parser list doesn't need to concern itself with what the types are of the individual parsers. We have a cast exception at runtime if it doesn't all line up :-)